### PR TITLE
Update versions for Azure Spring June release

### PR DESCRIPTION
**Pull request by backwind1233**

### CODE REVIEW

File: eng/jacoco-test-coverage/pom.xml [hunk 1]
```diff
@@ -410,47 +410,47 @@
     <dependency>
       <groupId>com.azure.spring</groupId>
       <artifactId>azure-identity-spring</artifactId>
-      <version>1.6.0-beta.1</version> <!-- {x-version-update;com.azure.spring:azure-identity-spring;current} -->
+      <version>1.6.0</version> <!-- {x-version-update;com.azure.spring:azure-identity-spring;current} -->
```
Comment (chenrujun) [inline eng/jacoco-test-coverage/pom.xml:413]: Do we need to add azure-security-keyvault-jca and azure-spring-boot-starter-keyvault-certificates in this pom.xml?

File: sdk/boms/azure-sdk-bom/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.0.3 (2021-06-02)
+## 1.0.4-beta.1 (2021-06-23)
```
Comment (chenrujun) [inline sdk/boms/azure-sdk-bom/CHANGELOG.md:3]: Don't change this.

File: sdk/keyvault/azure-security-test-keyvault-jca/pom.xml [hunk 3]
```diff
@@ -21,7 +21,7 @@
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-security-keyvault-jca</artifactId>
-      <version>1.0.0-beta.8</version> <!-- {x-version-update;com.azure:azure-security-keyvault-jca;current} -->
+      <version>1.0.0</version> <!-- {x-version-update;com.azure:azure-security-keyvault-jca;current} -->
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-test-keyvault-jca/pom.xml:24]: Maybe we should keep it to `1.0.0-beta.1`

File: sdk/spring/azure-spring-boot-samples/azure-appconfiguration-conversion-sample-complete/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Release History
 
-## 2.0.0-beta.1 (Unreleased)
+## 2.0.0-beta.1 (2021-06-23)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-appconfiguration-conversion-sample-complete/CHANGELOG.md:3]: Don't change this.

File: sdk/spring/azure-spring-boot-samples/azure-appconfiguration-conversion-sample-initial/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # Release History
 
-## 2.0.0-beta.1 (Unreleased)
+## 2.0.0-beta.1 (2021-06-23)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-appconfiguration-conversion-sample-initial/CHANGELOG.md:3]: Don't change this.

File: sdk/spring/azure-spring-boot-samples/azure-appconfiguration-sample/CHANGELOG.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
 
-## 2.0.0-beta.1 (Unreleased)
+## 2.0.0-beta.1 (2021-06-23)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-appconfiguration-sample/CHANGELOG.md:4]: Don't change this.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application.yaml [hunk 7]
```diff
@@ -2,29 +2,29 @@ spring:
   cloud:
     azure:
       eventhub:
-        connection-string: [eventhub-namespace-connection-string]
-        checkpoint-storage-account: [checkpoint-storage-account]
-        checkpoint-access-key: [checkpoint-access-key]
-        checkpoint-container: [checkpoint-container]
+        connection-string: Endpoint=sb://yiehtest.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=H3vdqCIDOarSap8IWoGrXcGZTiHGwjLcfEBL98Ntlkg=
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application.yaml:5]: Recover this file. And delete related resources.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: eng/jacoco-test-coverage/pom.xml (hunk 1 above)
Comment (backwind1233) [reply eng/jacoco-test-coverage/pom.xml:413]: I am not sure, I will look into it.
It's not related with spring release.

File: sdk/boms/azure-sdk-bom/CHANGELOG.md (hunk 2 above)
Comment (backwind1233) [reply sdk/boms/azure-sdk-bom/CHANGELOG.md:3]: Fixed

File: sdk/keyvault/azure-security-test-keyvault-jca/pom.xml (hunk 3 above)
Comment (backwind1233) [reply sdk/keyvault/azure-security-test-keyvault-jca/pom.xml:24]: can we?

File: sdk/spring/azure-spring-boot-samples/azure-appconfiguration-conversion-sample-initial/CHANGELOG.md (hunk 5 above)
Comment (backwind1233) [reply sdk/spring/azure-spring-boot-samples/azure-appconfiguration-conversion-sample-initial/CHANGELOG.md:3]: Fixed

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application.yaml (hunk 7 above)
Comment (backwind1233) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application.yaml:5]: Fixed

File: sdk/spring/azure-spring-boot-samples/azure-appconfiguration-conversion-sample-complete/CHANGELOG.md (hunk 4 above)
Comment (backwind1233) [reply sdk/spring/azure-spring-boot-samples/azure-appconfiguration-conversion-sample-complete/CHANGELOG.md]: Fixed

**Review by chenrujun — APPROVED** [review]
LGTM.

**Comment by chenrujun** [comment]
Hi, @backwind1233 , 
FYI: Issue created: https://github.com/Azure/azure-sdk-for-java/issues/22487

**Pull request merged**